pages/checkout/pay.vue

```diff
@@ -0,0 +1,85 @@
+<script lang="ts" setup>
+import { storeToRefs } from 'pinia';
+import { useMainStore } from '~/store';
+
+const { t } = useI18n();
+
+const route = useRoute();
+const store = useMainStore();
+const { plans } = storeToRefs(store);
+
+const steps = [
+  {
+    title: t('home.plans.tiers.step_1.title'),
+    description: t('home.plans.tiers.step_1.description'),
+    names: ['checkout-pay'],
+  },
+  {
+    title: t('home.plans.tiers.step_2.title'),
+    description: t('home.plans.tiers.step_2.description'),
+    names: ['checkout-pay-method'],
+  },
+  {
+    title: t('home.plans.tiers.step_3.title'),
+    description: t('home.plans.tiers.step_3.description'),
+    names: [
+      'checkout-pay-request-crypto',
+      'checkout-pay-crypto',
+      'checkout-pay-card',
+      'checkout-pay-paypal',
+    ],
+  },
+];
+
+const step = computed(() => {
+  const index = steps.findIndex((step) =>
+    step.names.includes(String(route.name)),
```

Comment:
does this behave different than `.toString()`

Reply:
it shouldn't

Reply:
it actually does, `route.name` can be `undefined`.
would require more logic to accomplish (eg route.name?.toString() ?? '')

Reply:
😞 javascript...